lib/glific/templates/interactive_templates.ex

```diff
@@ -113,17 +113,77 @@ defmodule Glific.Templates.InteractiveTemplates do
   @spec create_interactive_template(map()) ::
           {:ok, InteractiveTemplate.t()} | {:error, Ecto.Changeset.t()}
   def create_interactive_template(attrs) do
-    case validate_interactive_content_length(attrs) do
+    case contains_markdown_syntax?(attrs) do
       :ok ->
-        %InteractiveTemplate{}
-        |> InteractiveTemplate.changeset(attrs)
-        |> Repo.insert()
+        case validate_interactive_content_length(attrs) do
+          :ok ->
+            %InteractiveTemplate{}
+            |> InteractiveTemplate.changeset(attrs)
+            |> Repo.insert()
+
+          {:error, message} ->
+            {:error, message}
+        end
 
       {:error, message} ->
         {:error, message}
     end
```

Comment:
Can we use `with` clause instead of nested case?. Also when we use `with` we should make sure that we don't fall into [complex else clauses in with](https://hexdocs.pm/elixir/main/code-anti-patterns.html#complex-else-clauses-in-with) antipattern. In short here we are returning the error as it is in ln: 125 and ln:129. The mentioned doc provides solution to refactor this code. Can you check it?